src/amo/components/SearchForm.js

```diff
@@ -25,10 +31,10 @@ export class SearchFormBase extends React.Component {
   }
 
   goToSearch(query) {
-    const { api, pathname } = this.props;
+    const { addonType, api, pathname } = this.props;
     this.context.router.push({
       pathname: `/${api.lang}/${api.clientApp}${pathname}`,
-      query: { q: query },
+      query: { q: query, type: addonType },
```

Comment:
> Won't this push ?q=searchTerm&viewContext=extension? That's not what we want.

Yes, why not? I think `viewContext` makes more sense than `addonType` because it's only used to indicate the view context. It would make the code that parses the query string way more straight forward IMO.